src/devices/sound/pokey.cpp

```diff
@@ -1124,30 +1187,39 @@ void pokey_device::vol_init()
 
 }
 
-void pokey_device::poly_init_4_5(uint32_t *poly, int size, int xorbit, int invert)
+void pokey_device::poly_init_4_5(uint32_t *poly, int size)
 {
 	int mask = (1 << size) - 1;
 	int i;
 	uint32_t lfsr = 0;
 
 	LOG_POLY(("poly %d\n", size));
-	for( i = 0; i < mask; i++ )
+
+	if(size == 4)
 	{
-		/* calculate next bit */
-		int in = !((lfsr >> 0) & 1) ^ ((lfsr >> xorbit) & 1);
-		lfsr = lfsr >> 1;
-		lfsr = (in << (size-1)) | lfsr;
-		*poly = lfsr ^ invert;
-		LOG_POLY(("%05x: %02x\n", i, *poly));
-		poly++;
+		for( i = 0; i < mask; i++ )
+		{
+			lfsr = (lfsr+lfsr) + (~((lfsr >> 2) ^ (lfsr >> 3)) & 1);
+			*poly = lfsr & mask;
+			poly++;
+		}
+	}
+	else // size == 5
+	{
+		for( i = 0; i < mask; i++ )
+		{
+			lfsr = (lfsr+lfsr) + (~((lfsr >> 2) ^ (lfsr >> 4)) & 1);
+			*poly = lfsr & mask;
+			poly++;
+		}
 	}
```

Comment:
Can we simplify this if/else section and get rid of the identical loops? Something like:

```
int const xorbit = size - 1;
for (i = 0; i < mask; i++)
{
    lfsr = (lfsr << 1) | (~((lfsr >> 2) ^ (lfsr >> xorbit)) & 1);
    *poly = lfsr & mask;
    poly++;
} 
```

Note I got rid of the `+` operators in the first line of the `for` body too, which don't really feel idiomatic here for a LFSR.